backend/src/connectors/v2/authorisation/base.ts

```diff
@@ -0,0 +1,336 @@
+      }),
+    )
+  }
+
+  async files(
+    user: UserDoc,
+    model: ModelDoc,
+    files: Array<FileInterfaceDoc>,
+    action: FileActionKeys,
+  ): Promise<Array<Response>> {
+    // Does the user hold a role on the file's model?
+    const hasModelRole = (await authentication.getUserModelRoles(user, model)).length !== 0
+
+    // Does the user have a valid access request for this model?
+    const hasAccessRequest = await this.hasAccessRequest(user, model)
+
+    return Promise.all(
+      files.map(async (file) => {
+        // If they are not listed on the model, don't let them upload or delete files.
+        if (!hasModelRole && [FileAction.Delete, FileAction.Upload].includes(action)) {
+          return { success: false, info: 'You need to hold a model role in order to upload a file', id: file.id }
+        }
+
+        if (
+          !hasAccessRequest &&
+          !hasModelRole &&
+          [FileAction.Download].includes(action) &&
+          !model.settings.ungovernedAccess
+        ) {
+          return { success: false, info: 'You need to have a valid access request to download a file.', id: file.id }
+        }
+
+        return { success: true, id: file.id }
+      }),
+    )
+  }
+
+  async images(user: UserDoc, model: ModelDoc, accesses: Array<Access>): Promise<Array<Response>> {
+    // Does the user hold a role on the image's model?
+    const hasModelRole = (await authentication.getUserModelRoles(user, model)).length !== 0
+
+    // Does the user have a valid access request for this model?
+    const hasAccessRequest = await this.hasAccessRequest(user, model)
+
+    return Promise.all(
+      accesses.map(async (access) => {
+        // Don't allow anything beyond pushing and pulling actions.
+        if (
+          !access.actions.every((action) => [ImageAction.Push, ImageAction.Pull, ImageAction.List].includes(action))
+        ) {
+          return {
+            success: false,
+            info: 'You are not allowed to complete any actions beyond `push` or `pull` on an image.',
+            id: access.name,
+          }
+        }
+
+        // If they are not listed on the model, don't let them upload or delete images.
+        if (!hasModelRole && access.actions.includes(ImageAction.Push as Action)) {
+          return { success: false, info: 'You need to hold a model role in order to upload an image', id: access.name }
+        }
+
+        if (
+          !hasAccessRequest &&
+          !hasModelRole &&
+          access.actions.includes(ImageAction.Pull as Action) &&
+          !model.settings.ungovernedAccess
+        ) {
+          return {
+            success: false,
+            info: 'You need to have a valid access request to download an image.',
+            id: access.name,
+          }
+        }
+
+        return { success: true, id: access.name }
+      }),
+    )
+  }
+}
+
+// type Response = { success: true; info: undefined } | { success: false; info: string }
```

Comment:
Is there a reason this is commented out? If so can we add a "todo" to this part of the code, ideally with a Jira ticket if possible.